beacon_node/http_api/src/lib.rs

```diff
@@ -1539,30 +1592,47 @@ pub fn serve<T: BeaconChainTypes>(
             },
         );
 
-    // GET debug/beacon/heads
-    let get_debug_beacon_heads = eth1_v1
+    // GET v1/debug/beacon/heads
+    let get_debug_beacon_heads = any_version
         .and(warp::path("debug"))
         .and(warp::path("beacon"))
         .and(warp::path("heads"))
         .and(warp::path::end())
         .and(chain_filter.clone())
-        .and_then(|chain: Arc<BeaconChain<T>>| {
-            blocking_json_task(move || {
-                let heads = chain
-                    .heads()
-                    .into_iter()
-                    .map(|(root, slot)| api_types::ChainHeadData { slot, root })
-                    .collect::<Vec<_>>();
-                Ok(api_types::GenericResponse::from(heads))
-            })
-        });
+        .and_then(
+            |endpoint_version: EndpointVersion, chain: Arc<BeaconChain<T>>| {
+                blocking_json_task(move || {
+                    let heads = chain
+                        .heads()
+                        .into_iter()
+                        .map(|(root, slot)| {
+                            let execution_optimistic = if endpoint_version == V1 {
+                                None
+                            } else if endpoint_version == V2 {
+                                BlockId::from_root(root)
+                                    .is_execution_optimistic(&chain)
+                                    .ok()
```

Comment:
Note that this code will return `execution_optimistic: None` upon an error (which will then be `skip_serialized` out of the response).
I wanted to avoid the case where there are multiple heads but an error in one prevented any response from being shown. This felt like the lesser of two evils. The downside is that this behavior is not very clear. As this is a debug endpoint, this might be fine however.

It seems to me that there are two options here:
1. An error for 1 head is an error for all heads
2. No errors, but `execution_optimistic: None` means that there was an error in it's computation.
(This should be distinct since to get this new behaviour you have to use the v2 endpoint)

Happy to hear alternate solutions.